### DISCUSSION TITLE
IMPLEMENT ADD NEW COMPONENT FEATURE

### DISCUSSION THREAD

**Discussion by QuangLanh** [clientIO/joint#2676] [Q&A]
### Introduction

On UI, can users add new components. 
For example, clicking the Plus icon will allow users to select components under their device and add them.
![image](https://github.com/clientIO/joint/assets/91791283/21357afc-eaff-4a9b-a0c7-05f63186b418).
Please reply me early, thanks.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
The [Stencil theme picker](https://www.jointjs.com/demos/stencil-theme-picker) example shows how to add a button to the stencil group header.

You can add elements to the stencil dynamically using [`stencil.loadGroup()`](https://docs.jointjs.com/api/ui/Stencil/#loadgroup).
